---
title: "makeCountMatrix"
author: "Rylee Hackley"
date: "OCT 2023"
output: github_document
---

```{r, message=FALSE}
library(tidyverse)
library(magrittr)
library(purrr)
library(DESeq2)
```

```{r, message=FALSE, warning=FALSE}
#read all counts (from 2019) into single dataframe
counts2019 <- read_tsv("00_counts/2019_counts.txt", skip = 1)
tmp <- colnames(counts2019)
tmp2 <- str_replace_all(tmp, "-", "_")
tmp3 <- str_remove(tmp2, "_sorted.bam")
colnames(counts2019) <- tmp3

## remove counts with less than 60% alignment
tmp <- c("dtbsP_4hmfe_5_S46", "dtbsP_4hmfe_2_S40", "dtrmBdtbsP_4hmfe_2_S56", "dtrmBdtbsP_4hpfe_2_S54", "dtrmBdtbsP_4hpfe_6_S69")

counts2019 %<>% select(!tmp)

#check genotype matches label:
counts2019 %>%
  filter(Geneid == "HVO_RS18535" | Geneid == "HVO_RS17680")

#remove mislabeled samples
tmp <- c("dtbsP_4hmfe_6_S52", "dtrmBdtbsP_4hpfe_5_S65") #(have tbsP and trmB expr when shouldn't)
counts2019 %<>% select(!tmp)

counts2019wt <- read_tsv("00_counts/2019_counts2.txt", skip = 1)[16]
colnames(counts2019wt) <- str_replace_all(colnames(counts2019wt), "-", "_")
colnames(counts2019wt) <- str_remove(colnames(counts2019wt), "_L002_R1_001.fastq.gz_sorted.bam")

counts2019 <- cbind(counts2019, counts2019wt)
```

```{r, message=F, warning=FALSE}
counts2021 <- read_tsv("00_counts/2021_counts.txt", skip = 1)

#drop mislabeled hvo sample in column 15:
counts2021 <- counts2021[-15]
colnames(counts2021)[7] <- "HVO_DS2_reference"

colnames(counts2021) <- str_remove(colnames(counts2021) , "_sorted.bam")

#check genotype matches label:
counts2021 %>%
  filter(Geneid == "HVO_RS18535" | Geneid == "HVO_RS17680")
```


```{r}
#correct swapped trmB_a_glu and tbsP_e_glu samples: 
a <- colnames(counts2021)[23]
b <- colnames(counts2021)[25]

colnames(counts2021)[23] <- b
colnames(counts2021)[25] <- a

#merge
(left_join(counts2021, counts2019, by = c("Geneid", "Chr", "Start", "End", "Strand", "Length")) -> counts_combined)

#calc total reads per sample
as.data.frame(colSums(counts_combined[-c(1:6)])) -> cnt_sums
cnt_sums <- rownames_to_column(cnt_sums, "sample_name")

#export combined counts
write_csv(counts_combined, "00_counts/00_combined_counts.csv")
write_csv(counts_combined, "00_deseq2_input/00_combined_data.csv")
write_csv(cnt_sums, "00_counts/00_counts_sums.csv")
```
